crates/telio-starcast/src/nat.rs

```diff
@@ -0,0 +1,744 @@
+use crate::utils::{DualIpAddr, MutableIpPacket};
+use pnet_packet::{
+    ip::IpNextHeaderProtocols, ipv4::MutableIpv4Packet, ipv6::MutableIpv6Packet,
+    udp::MutableUdpPacket,
+};
+use std::{
+    hash::{DefaultHasher, Hash, Hasher},
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+    time::Duration,
+};
+use telio_utils::{telio_log_warn, LruCache};
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// The packet is too short to be a valid IP packet
+    PacketTooShort,
+    /// The packet has an unexpected IP version
+    UnexpectedIpVersion,
+    /// The packet has an unexpected transport protocol
+    UnexpectedTransportProtocol,
+    /// Cannot map the packet
+    CannotMap,
+    /// There is no mapping for the given port
+    MappingNotFound,
+}
+
+pub trait Nat {
+    /// Translate incomming packet (from the transport socket to the multicast peer)
```

Comment:
Nit again: incoming :smile: There's plenty of the same typo in the tests in this file as well